collection+subdoc.go

```diff
@@ -145,6 +145,30 @@ func evalSubdocPath(subdoc any, path []string) (any, error) {
 	return subdoc, nil
 }
 
+// Upserts the value at the specified JSON path in the source.
+// Returns ErrPathMismatch if non-leaf path entries do not exist.
+func upsertSubdocValue(source any, path []string, value interface{}) error {
+
+	// eval path exists
+	subdoc, err := evalSubdocPath(source, path[0:len(path)-1])
```

Comment:
readability nit:

I'd name `numLeaves := len(path) -1` because we use it again.

Reply:
I evaluated this, but prefer leaving as len(path)-1 as I feel `lastPath := path[len(path)-1]` is more readable than having to go back to see what numLeaves is.